src/app/app-routing.module.ts

```diff
@@ -1,7 +1,79 @@
 import { NgModule } from "@angular/core";
+import { LayoutComponent } from "./layout/layout.component";
+import { HomeComponent } from "./home/home.component";
+import { ErrorComponent } from "./error/error.component";
+import { LibrarianComponent } from "./librarian/librarian.component";
+import { BorrowerComponent } from "./borrower/borrower.component";
+import { AuthorComponent } from "./administrator/author/author.component";
+import { AdministratorComponent } from "./administrator/administrator.component";
+import { BookComponent } from "./administrator/book/book.component";
+import { BranchComponent } from "./administrator/branch/branch.component";
+import { PublisherComponent } from "./administrator/publisher/publisher.component";
+import { GenreComponent } from "./administrator/genre/genre.component";
+import { OverrideComponent } from "./administrator/override/override.component";
 import { Routes, RouterModule } from "@angular/router";
 
-const routes: Routes = [];
+const routes: Routes = [
+  {
+    path: "",
+    component: LayoutComponent,
+    children: [
+      {
+        path: "",
+        redirectTo: "lms/home",
+        pathMatch: "full",
+      },
+      {
+        path: "lms/home",
+        component: HomeComponent,
+      },
+      {
+        path: "lms/admin",
+        component: AdministratorComponent,
+      },
+      {
+        path: "lms/admin/author",
+        component: AuthorComponent,
+      },
+      {
+        path: "lms/admin/book",
+        component: BookComponent,
+      },
+      {
+        path: "lms/admin/borrower",
+        component: BorrowerComponent,
```

Comment:
This goess to the component that borrowers use to checkout & return books, not the admin's borrower component. We should probably give the admin's borrower component a different name, like AdminBorrowerComponent or BorrowerCrudComponent. Then to be consistent we should probably give similar names to the other admin components, other than override.

Reply:
I added the admin prefix where necessary. good catch!